File: library/cpp/tvmauth/client/examples/create_with_tvmtool/create.cpp
#include <library/cpp/tvmauth/client/facade.h> 
 
namespace NExample { 
    // Possibility of using functions depends on config of tvmtool 
    //    CheckServiceTicket 
    //    CheckUserTicket 
    //    GetServiceTicketFor 
 
    NTvmAuth::TTvmClient CreateClientInQloudOrYandexDeploy() { 
        NTvmAuth::NTvmTool::TClientSettings setts( 
            "my_service" // specified in Qloud/YP/tvmtool interface 
        ); 
 
        NTvmAuth::TLoggerPtr log = MakeIntrusive<NTvmAuth::TCerrLogger>(7); 
 
        NTvmAuth::TTvmClient c(setts, log); 
 
        return c; 
    } 
 
    NTvmAuth::TTvmClient CreateClientForDevOrTests() { 
        NTvmAuth::NTvmTool::TClientSettings setts( 
            "my_service" // specified in Qloud/YP/tvmtool interface 
        ); 
        setts.SetPort(18080); 
        setts.SetAuthToken("AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"); 
 
        NTvmAuth::TLoggerPtr log = MakeIntrusive<NTvmAuth::TCerrLogger>(7); 
 
        NTvmAuth::TTvmClient c(setts, log); 
 
        return c; 
    } 
} 

File: library/java/tvmauth/examples/create_with_tvmtool/Create.java
package ru.yandex.example;

import ru.yandex.passport.tvmauth.NativeTvmClient;
import ru.yandex.passport.tvmauth.TvmClient;
import ru.yandex.passport.tvmauth.TvmToolSettings;

public class Create {
    // Possibility of using functions depends on config of tvmtool
    //    checkServiceTicket
    //    checkUserTicket
    //    getServiceTicketFor

    public static TvmClient createClientInQloudOrYandexDeploy() {
        TvmToolSettings settings = TvmToolSettings.create("my_service");

        NativeTvmClient c = new NativeTvmClient(settings);

        return c;
    }

    public static TvmClient createClientForDevOrTests() {
        TvmToolSettings settings = TvmToolSettings.create("my_service");
        settings.setPort(18080);
        settings.setAuthToken("AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA");

        NativeTvmClient c = new NativeTvmClient(settings);

        return c;
    }
}
